src/storage/admin/AdminTaskManager.cpp

```diff
@@ -25,10 +26,83 @@ bool AdminTaskManager::init() {
 
   bgThread_->addTask(&AdminTaskManager::schedule, this);
   shutdown_ = false;
+  handleUnreportedTasks();
   LOG(INFO) << "exit AdminTaskManager::init()";
   return true;
 }
 
+void AdminTaskManager::handleUnreportedTasks() {
+  if (env_ == nullptr) return;
+  unreportedAdminThread_.reset(new std::thread([this] {
+    bool ifAny = true;
+    while (ifAny) {
+      ifAny = false;
+      std::unique_ptr<kvstore::KVIterator> iter;
+      auto kvRet = env_->adminStore_->scan(&iter);
+      if (kvRet != nebula::cpp2::ErrorCode::SUCCEEDED || iter == nullptr) return;
+      std::vector<std::string> keys;
+      for (; iter->valid(); iter->next()) {
+        folly::StringPiece key = iter->key();
+        int32_t seqId = *reinterpret_cast<const int32_t*>(key.data());
+        if (seqId >= env_->adminSeqId_ || seqId < 0) continue;
+        JobID jobId = *reinterpret_cast<const JobID*>(key.data() + sizeof(int32_t));
+        TaskID taskId =
+            *reinterpret_cast<const TaskID*>(key.data() + sizeof(int32_t) + sizeof(JobID));
+        folly::StringPiece val = iter->val();
+        nebula::meta::cpp2::JobStatus jobStatus =
+            *reinterpret_cast<const nebula::meta::cpp2::JobStatus*>(val.data() +
+                                                                    sizeof(meta::cpp2::AdminCmd));
+        nebula::cpp2::ErrorCode errCode;
+        nebula::meta::cpp2::StatsItem statsItem;
+        if (jobStatus != nebula::meta::cpp2::JobStatus::FINISHED) {
+          errCode = nebula::cpp2::ErrorCode::E_TASK_EXECUTION_FAILED;
+          statsItem.set_status(nebula::meta::cpp2::JobStatus::FAILED);
+        } else {
+          errCode = nebula::cpp2::ErrorCode::SUCCEEDED;
+          statsItem.set_status(nebula::meta::cpp2::JobStatus::FINISHED);
+        }
+        meta::cpp2::StatsItem* pStats = &statsItem;
+
+        LOG(INFO) << folly::sformat("reportTaskFinish(), job={}, task={}, rc={}",
+                                    jobId,
+                                    taskId,
+                                    apache::thrift::util::enumNameSafe(errCode));
+        auto rc = nebula::cpp2::ErrorCode::SUCCEEDED;
+        auto fut = env_->metaClient_->reportTaskFinish(jobId, taskId, errCode, pStats);
+        fut.wait();
```

Comment:
It seems that there is no need to block the above asynchronous tasks. Why not use the future chains to make it asynchronous execution?